### PR TITLE
feat: add GRPC server and JSON RPC server

### PR DESCRIPTION
**Pull request by adu-web3**
## Description

front end needs to fetch the validator container data, withdrawal container data and also the proof data to build the tx that calls `depositBeaconChainValidator` or `processWithdrawal`, so that we need to add grpc server and json server to enable this feature

## Summary by CodeRabbit

- **New Features**
  - Introduced new configuration files for managing Protocol Buffers and dependencies.
  - Added structured JSON files for validator and withdrawal proofs, enhancing blockchain state verification.
  - Implemented a `ProofServer` for generating validator and withdrawal proofs, supporting Ethereum 2.0 functionalities.
  - Established a dual communication capability with a gRPC server and an HTTP server.

- **Bug Fixes**
  - Improved error handling and logging during proof generation processes.

- **Tests**
  - Added unit tests to validate new JSON structures and proof functionalities.

### CODE REVIEW

**Comment by adu-web3** [comment]
I deploy this on qa-test machine, which has an beacon node that exposes rpc locally and it works as expected:

```bash
curl -X POST -H "Content-Type: application/json" -d '{
>   "slot": 2300000,
>   "validator_index": 1647525
> }' http://localhost:8989/v1/validator-proof
{"stateRoot":"oad7h+JTJLob7kuKcKXEcmAABwn1cIUNJlJtfaS+uC4=", "stateRootProof":["hYFbKzjahiLeHOkQwwet7KldrbIryShnyQfEFMv3Iuo=", "Lu24jFQpF6iDssWpKu8MQzdMPDYkC5scqcAeMlx/X+4=", "KHr1fjYkIQFAp2XsXs4p7J1LHFyBo0vKm3aRwWsRRgk="], "validatorContainer":["0xed43e89d71901bd903479efa75ec36640777119e7c023675a8ed72b1d9ef82b8", "0x010000000000000000000000b87ccabe08275cc5293339589af7ac69f20c06b6", "0x0040597307000000000000000000000000000000000000000000000000000000", "0x0000000000000000000000000000000000000000000000000000000000000000", "0xada0000000000000000000000000000000000000000000000000000000000000", "0x67a8000000000000000000000000000000000000000000000000000000000000", "0xffffffffffffffff000000000000000000000000000000000000000000000000", "0xffffffffffffffff000000000000000000000000000000000000000000000000"], "validatorContainerProof":["+FpULyP7+gODjunLteZVL/6SUyzpaNw+H/4Bt+YwoC0=", "bJLkJEfveRz02vt3qYH5fQnxSoZ/5rN/dCP/GdRC0mc=", "pQFjlA9hySCftDkk/ZQqOLCkw1msmpdyGZ2sixBAuCo=", "3JcCg/2ieJgwQ5eQTl5CVY+vCAz7x/OCDswpeODDDQM=", "q6Ri4HkczkkEYfd3EfogZ9Pls4Aab6mgQReBRJth3oY=", "34hFS/CpLI8HQILko7FxHcb/WRQnG/yV6I78LhN1oMI=", "GrjVTLBvrrmgshFhrD3hl5SvcBNWWRhE5gPsGM/UioI=", "n4+DJKJfWNTK7wQe10x7YytjBHBMl0CRqnFXr11NyhI=", "roQbfk8gmuim1BY11Xbc4l5+ZF65tq4pxsMF0+rBn8w=", "xMMygUYPaikeBysvJiz+fenxgWphT9Vaq7o2Vubbt8A=", "1uDFC3fNqmfsR2J7d+qnYywC01+pccvo+NkEt1293hg=", "NkMJf3etA3sAvxis0DTXtm7RhoRIUKOGwS8pjSQSu24=", "n+aEhJvW67AwFCNYeDAeCrF2U02Gq24/ZTrHAJ2HF2k=", "uLnLhGV4kEZE8lU6joFmM7DAG/w9PnlMP2+wphNwpYI=", "JwbG1zGt8ltUN/u1m8I3LtQXPCqx6rlpXrMLe7zovd0=", "tG5lD1yw5tUKAzkPTL1GF+J1ZZez63qmqOICPmo0Gh0=", "mox1f1b2a42+t8/aZ8S2L6ZsJKrjhE2X3J9oG8FSzW0=", "j2zZbnUcyZgmv7qVwZgYYvo51JpMdefUuFAzWo8thJ8=", "le7IsuVBytTpHeODhfLgRmGfVElsI4LLbKzVuYwm9aQ=", "wCk0K9EzNdStYVAwg1s9S7zs6F5/ix1Bpylfk/CXlTU=", "VOX636KqcSFtSA+sACTCbyHDJGEl6P54LV59pnHhT+g=", "io1/46+MqghadjmoMgAUV9+5EoqAYRQq0DNWKf8j/5w=", "/rPDN9elGm+/ALnjTFLhyRlclpvU56C/1R1cW+2cEWc=", "5x8KqDzDLt+++p9NPgF0yoUYLuyfOgn2psDfY3elENc=", "MSBvqApQu2q+KQhQWPFiEiEqYO7I8En+y5LYyOCoS8A=", "ITUr/svt3emTg59hTD2sCj7jdUP5tBKxYZncFY4jtUQ=", "YZ4xJyS7bXwxU+2d55HXZKNms4mvE8WL+KjZBIGkZ2U=", "fN0phiaCUGKNDBDjhcWMYZHm++BRkbzATxM/LOpywcQ=", "hIkwvXuoysVGYQchE/sniGnge7hYf5E5KTM3TQF7y+E=", "iGn/LCKyjMEFENmFMpKAMyi+T7DoBJXou40nH1uIljY=", "tf4o558bhQ+GWCRs6bah57SfwG23FD6P4LTysMVSOlw=", "mF6Sn3CvKNC90akKgI+Xf1l8fHeMSJ6Y072JENMawPc=", "xvZ+Aubk4b3vuZTGCYlT80Y2uitsogpHIdKyaohnIv8=", "HJp+X/HPSLStFYLT9OShAE87INjForcTh6QlStkz68U=", "Lwda4ilka29q7Rml43LPKVCBQB64k/9Zmz+azAwNPn0=", "Mokh3rWWEgdoAejNYVkhB7XGfHm4RllcxjIMOVtGNiw=", "v7kJ/bI2rSQRtOSIOBCgdLhARkaJmGw/ioCRgn4Xwyc=", "Vdj7Noe6O6SfNCx39aH4m+yD2BFEbhpGcTkhPWQLanQ=", "9yENT45+EDl5Dnv076IHVVoQptsd1LldoxOqqIuI/nY=", "rSG1FsvGRf/jSrXeHIrvjNTn+NK1Ho4UVq3HVjzaIG8=", "eGYbAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=", "yXsFAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=", "277JapZ+sf6MazcOIUeDAPaPLACbMhQciY0MQe0YdWA=", "qHPQ2HxhFgi7NxwmYl5vKcUU4nBpoKuJwiG34pYjMMM=", "1LpOSP5PQIZD/NFeWot7Uy7JWPeh8QVhLSnSZmgsPhw=", "uAsVW3QoNUTv8BQZKDwqS/jTWF7zETAhy+Z4qh/WQdA="]}
```

File: common_utils/utils.go [hunk 1]
```diff
@@ -148,10 +149,23 @@ type InputDataBlock struct {
 	Finalized            bool `json:"finalized"`
 }
 
-func ConvertBytesToStrings(b [][32]byte) []string {
+func ConvertBytesToStrings(b interface{}) []string {
 	var s []string
-	for _, v := range b {
-		s = append(s, "0x"+hex.EncodeToString(v[:]))
+	switch v := b.(type) {
+	case [][]byte:
```
Comment (bwhour) [inline common_utils/utils.go:155]: It looks like using type merging can simplify the original code.

> case [][]byte, [][32]byte, common.Proof:
    for _, bytes := range v {
        s = append(s, "0x"+hex.EncodeToString(bytes[:]))
    }

File: server/main.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	err = yaml.Unmarshal(data, &config)
+	if err != nil {
+		panic(fmt.Sprintf("Error parsing config file: %v", err))
+	}
+
+	return config
+}
+
+func startGRPCServer(config Config) {
+	lis, err := net.Listen("tcp", ":"+config.GRPCPort)
```
Comment (bwhour) [inline server/main.go:52]: defer lis.Close()

**Comment by cloud8little** [comment]
https://github.com/ExocoreNetwork/proofs-generation/pull/1/commits/cfb825f0342f5dea0aed9335d0090f7922ec9f47  Test passed.

**Review by cloud8little — APPROVED** [review]

File: common_utils/utils.go (hunk 1 above)
Comment (adu-web3) [reply common_utils/utils.go:155]: I tried, but it reports error for common.Proof

**Review by bwhour — APPROVED** [review]

**Pull request merged**